### PR TITLE
Transition to SeawaterPolynomials

### PR DESCRIPTION
**Pull request by glwagner**
This PR removes the `RoquetEquationOfState` and `TEOS10` and adds a dependency to `SeawaterPolynomials`.

We need

- [x] remove the tests that depend on Roquet and TEOS10
- [x] test that we can time-step a model with a `SeawaterPolynomials.BoussinesqEquationOfState`

### CODE REVIEW

**Comment by glwagner** [comment]
The tests are erroring during the docs build. I don't know why. Is it because `SeawaterPolynomials` is not yet a package? We are still waiting for approval from `JuliaRegistries/General`.

**Comment by glwagner** [comment]
Finally tests pass! @ali-ramadhan ?

File: Manifest.toml [hunk 1]
```diff
@@ -255,6 +249,14 @@ version = "1.0.1"
 [[SHA]]
 uuid = "ea8e919c-243c-51af-8825-aaa63cd721ce"
 
+[[SeawaterPolynomials]]
+deps = ["Test"]
+git-tree-sha1 = "3719c0c269f679a39332633acf812ce2dbf8aaa8"
+repo-rev = "master"
+repo-url = "https://github.com/glwagner/SeawaterPolynomials.jl.git"
+uuid = "d496a93d-167e-4197-9f49-d3af4ff8fe40"
+version = "0.2.2"
```
Comment (ali-ramadhan) [inline Manifest.toml:258]: Looks like SeawaterPolynomials.jl is a Julia package now but we should probably clean things up so the releases are tagged?

https://github.com/JuliaRegistries/General/pull/13734#issuecomment-621936685

Comment (glwagner) [reply Manifest.toml:258]: I like bumping version numbers whenever possible.

**Pull request merged**